admin/database/postgres/postgres.go

```diff
@@ -132,3 +133,95 @@ func (c *connection) DeleteProject(ctx context.Context, id string) error {
 	_, err := c.db.ExecContext(ctx, "DELETE FROM projects WHERE id=$1", id)
 	return err
 }
+
+// InsertAuthCode inserts the authorization code data into the store.
+func (c *connection) InsertAuthCode(ctx context.Context, code *database.AuthCode) error {
+	res, err := c.db.ExecContext(ctx,
+		"INSERT INTO device_code_auth (device_code, user_code, expires_at, approval_state, client_id, user_id, used) VALUES ($1, $2, $3, $4, $5, $6, $7)",
+		code.DeviceCode, code.UserCode, code.Expiry, code.ApprovalState, code.ClientID, code.UserID, code.Used)
+	if err != nil {
+		return err
+	}
+	rows, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if rows != 1 {
+		return fmt.Errorf("could not insert auth code, expected 1 row to be affected, got %d", rows)
+	}
```

Comment:
was not sure of the semantics here so added a check just in case, can remove the check from inserts. will keep for `update` and `delete`, does that make sense ?

Reply:
Sounds good, please add in other update/delete handlers calls as well. Also, instead of returning a custom error, return `database.ErrNotFound` (so it's easier to check for in caller).